### PR TITLE
Popcorn auth

### PR DESCRIPTION
**Pull request by Morriar**
## Authentification handlers.

For now, only Github OAuth is provided.

### Github

See https://developer.github.com/v3/oauth/.

This module provide 4 base classes that can be used to implement a Github
OAuth handshake.

~~~nit
import popcorn
import popcorn::pop_auth

var client_id = "github client id"
var client_secret = "github client secret"

var app = new App
app.use("/login", new GithubLogin(client_id))
app.use("/oauth", new GithubOAuthCallBack(client_id, client_secret))
app.use("/logout", new GithubLogout)
app.listen("localhost", 3000)
~~~

Optionaly, you can use the `GithubUser` handler to provide access to the
Github user stored in session:

~~~nit
app.use("/api/user", new GithubUser)
~~~

### CODE REVIEW

File: lib/github/api.nit [hunk 1]
```diff
@@ -145,6 +145,19 @@ class GithubAPI
 		return res.as(JsonObject)
 	end
 
+	# Get the Github logged user from `auth` token.
+	#
+	# Loads the `User` from the API or returns `null` if the user cannot be found.
+	#
+	#     var api = new GithubAPI(get_github_oauth)
+	#     var user = api.load_auth_user
+	#     assert user.login == "Morriar"
```
Comment (privat) [inline lib/github/api.nit]: nitish? Because `get_github_oauth` will likely not login with `Morriar` on test machines

**Comment by privat** [comment]
I'm not sure I really understand how to identify/use the user once the login is done on the github side

**Comment by privat** [comment]
What about a simple example that login with github then says 'hello {name}'?

**Comment by Morriar** [comment]
Reroll with more examples in the documentation.

**Comment by privat** [comment]
+1

**Comment by privat** [comment]
test this please

**Pull request merged**